pkg/diagnostics/analyzers_test.go

```diff
@@ -3,18 +3,20 @@ package diagnostics_test
 import (
 	"testing"
 
-	"github.com/stretchr/testify/assert"
+	. "github.com/onsi/gomega"
 
+	eksav1alpha1 "github.com/aws/eks-anywhere/pkg/api/v1alpha1"
 	"github.com/aws/eks-anywhere/pkg/diagnostics"
 )
 
 func TestManagementClusterAnalyzers(t *testing.T) {
+	g := NewGomegaWithT(t)
 	factory := diagnostics.NewAnalyzerFactory()
 	analyzers := factory.ManagementClusterAnalyzers()
-	assert.Equal(t, len(analyzers), 12, "DataCenterConfigCollectors() mismatch between desired collectors and actual")
-	assert.NotNilf(t, getDeploymentStatusAnalyzer(analyzers, "capc-controller-manager"), "capc controller manager analyzer should be present")
-	assert.NotNilf(t, getDeploymentStatusAnalyzer(analyzers, "capv-controller-manager"), "capv controller manager analyzer should be present")
-	assert.NotNilf(t, getDeploymentStatusAnalyzer(analyzers, "capt-controller-manager"), "capt controller manager analyzer should be present")
+	g.Expect(len(analyzers)).To(Equal(12), "DataCenterConfigCollectors() mismatch between desired collectors and actual")
+	g.Expect(getDeploymentStatusAnalyzer(analyzers, "capc-controller-manager")).ToNot(Equal(nil), "capc controller manager analyzer should be present")
```

Comment:
why only nil check? aren't we able to predict the content of the structs? Happy to take this offline if you prefer!

Reply:
This test is obtained from previous work. It seems that the following `getDeploymentStatusAnalyzer()` tests the expected content by checking the `deploymentStatus.Name == name`. And the number of returned analyzers in each namespace is variable. I can add two more test regarding the crd content. Does it make sense?